Warnings on build

The build currently produces a handful a warnings. For example:

```
src/Language/Python/Common/ParseError.hs:25:10: warning: [-Wdeprecations]
    In the use of type constructor or class ‘Error’
    (imported from Control.Monad.Error.Class, but defined in Control.Monad.Trans.Error):
    Deprecated: "Use Control.Monad.Trans.Except instead"
   |                                 
25 | instance Error ParseError where 
   |          ^^^^^     
```

```
src/Language/Python/Common/ParserUtils.hs:98:1: warning: [-Wincomplete-patterns]
    Pattern match(es) are non-exhaustive
    In an equation for ‘makeTupleOrExpr’: Patterns not matched: [] _
   |                                 
98 | makeTupleOrExpr [e] Nothing = e 
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
```


They mostly look pretty simple to fix; I've [started a branch](https://github.com/dperelman/language-python/tree/fix/warnings) where I'm trying to do so. But I'm not super-familiar with Haskell style, so I'm not sure exactly how to properly fix those warnings or maybe if simply telling the compiler to not warn for those things is the proper fix.

(My impetus for doing so was that when I saw the warnings, I first thought they might mean the package was actually broken because it was taking a long time to compile. After a few minutes, it made progress, so I realized they were not actually breaking anything.)

--- a/src/Language/Python/Common/LexerUtils.hs
+++ b/src/Language/Python/Common/LexerUtils.hs
@@ -14,7 +14,6 @@
 module Language.Python.Common.LexerUtils where
 
 import Control.Monad (liftM)
-import Control.Monad.Error.Class (throwError)
 import Data.List (foldl')
 import Data.Word (Word8)
 import Language.Python.Common.Token as Token

--- a/src/Language/Python/Common/ParseError.hs
+++ b/src/Language/Python/Common/ParseError.hs
@@ -14,14 +14,9 @@ module Language.Python.Common.ParseError ( ParseError (..) ) where
 
 import Language.Python.Common.SrcLocation (SrcLocation)
 import Language.Python.Common.Token (Token)
-import Control.Monad.Error.Class
 
 data ParseError  
    = UnexpectedToken Token           -- ^ An error from the parser. Token found where it should not be. Note: tokens contain their own source span.
    | UnexpectedChar Char SrcLocation -- ^ An error from the lexer. Character found where it should not be.
    | StrError String                 -- ^ A generic error containing a string message. No source location.
    deriving (Eq, Ord, Show)
-
-instance Error ParseError where
-   noMsg = StrError ""
-   strMsg = StrError 

--- a/src/Language/Python/Common/ParserMonad.hs
+++ b/src/Language/Python/Common/ParserMonad.hs
@@ -43,6 +43,7 @@ module Language.Python.Common.ParserMonad
    , addComment
    , getComments
    , spanError
+   , throwError
    ) where
 
 import Language.Python.Common.SrcLocation (SrcLocation (..), SrcSpan (..), Span (..))
@@ -51,7 +52,6 @@ import Language.Python.Common.ParseError (ParseError (..))
 import Control.Applicative ((<$>))
 import Control.Monad.State.Class
 import Control.Monad.State.Strict as State
-import Control.Monad.Error as Error
 import Language.Python.Common.Pretty
 
 internalError :: String -> P a 
@@ -90,6 +90,9 @@ initialState initLoc inp scStack
    }
 
 type P a = StateT ParseState (Either ParseError) a
+
+throwError :: ParseError -> P a
+throwError = lift . Left
 
 execParser :: P a -> ParseState -> Either ParseError a
 execParser = evalStateT 

--- a/src/Language/Python/Common/ParserUtils.hs
+++ b/src/Language/Python/Common/ParserUtils.hs
@@ -15,7 +15,6 @@ module Language.Python.Common.ParserUtils where
 
 import Data.List (foldl')
 import Data.Maybe (isJust)
-import Control.Monad.Error.Class (throwError)
 import Language.Python.Common.AST as AST
 import Language.Python.Common.Token as Token 
 import Language.Python.Common.ParserMonad hiding (location)

